airbyte-workers/src/test/java/io/airbyte/workers/DefaultReplicationWorkerTest.java

```diff
@@ -133,7 +134,7 @@ void test() throws Exception {
     verify(destination).start(destinationConfig, jobRoot);
     verify(destination).accept(RECORD_MESSAGE1);
     verify(destination).accept(RECORD_MESSAGE2);
-    verify(source).close();
+    verify(source, atLeastOnce()).close();
```

Comment:
I might not have understood this completely - in which scenario the source might be closed for more than once?

Reply:
ah good point.  I made a brief mention of this in the commit message [here](https://github.com/airbytehq/airbyte/pull/12632/commits/9831d347fe5c9f2b28bdea951eddd683eece23e2) but basically the source implements the AutoCloseable interface, and I am also explicitly calling close with this change, so close ends up being called twice